### PR TITLE
add withSearchResultsComponent

### PR DESCRIPTION
**Pull request by jacobvenable**
This PR does not close an issue.

## What does this PR do?

- creates a withSearchResults component that accepts a search query and makes a request to Algolia via React Hooks
- updates Algolia env variables to be prefaced with required `REACT_APP` ([required by CRA](https://facebook.github.io/create-react-app/docs/adding-custom-environment-variables))
- adds environment variable types

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,5 +1,13 @@
 import { SvgIconProps } from '@material-ui/core/SvgIcon';
 
+export interface TEnvVariables {
```
Comment (rhinodavid) [inline src/types.ts]: can you add the others

**Comment by jacobvenable** [comment]
> are you copying me in that gif

... maybe...

**Comment by jacobvenable** [comment]
Let's call it adhering to standards.

File: src/types.ts (hunk 1 above)
Comment (jacobvenable) [reply src/types.ts]: [shore thang](https://github.com/CodeForBoulder/upswyng/pull/78/commits/cfb821fb6807107cdc15eb6f2974412d9e5fa740)

**Pull request merged**